x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/CacheIteratorHelper.java

```diff
@@ -56,4 +56,17 @@ public void removeKeysIf(Predicate<K> removeIf) {
             }
         }
     }
+
+    public void removeValuesIf(Predicate<V> removeIf) {
```

Comment:
Where do we call this? I think we should avoid having this method if we can.

Reply:
This is indeed not needed anymore. Will remove.